lib/eslint.js

```diff
@@ -757,6 +757,8 @@ module.exports = (function() {
      */
     api.verify = function(textOrSourceCode, config, filenameOrOptions, saveState) {
         const text = (typeof textOrSourceCode === "string") ? textOrSourceCode : null;
+        const selectors = new Set();
```

Comment:
I decided to switch to this approach so that `eslint.on(selector, func)` would still work. The constructor feature-tests for `eventNames` beforehand, so I think it's probably fine to use it (if it gets removed in a future version, we will end up taking the `eventNames` code path anyway). See [this comment](https://github.com/eslint/eslint/blob/eada4ad09e9495e6c40a2598778e822ac03cea8c/lib/util/node-event-generator.js#L183-L197) for more explanation.